src/api/app/models/user.rb

```diff
@@ -554,12 +545,7 @@ def can_modify_project?(project, ignoreLock = nil)
     # we don't touch the cache for ignoreLock
     return can_modify_project_internal(project, ignoreLock) if ignoreLock
 
-    unless @projects_to_modify.has_key? project.id
-      # build cache
-      @projects_to_modify[project.id] = can_modify_project_internal(project, nil)
-    end
-
-    @projects_to_modify[project.id]
+    can_modify_project_internal(project, nil)
```

Comment:
Lines 546 and 548 are the same, so replace them with this one:
```
can_modify_project_internal(project, ignoreLock)
```
and also remove the comment in line 545